internals/overlord/overlord.go

```diff
@@ -183,6 +194,14 @@ func New(opts *Options) (*Overlord, error) {
 	// before it.
 	o.stateEng.AddManager(o.runner)
 
+	// Load the plan from the Pebble layers directory (which may be missing
+	// or have no layers, resulting in an empty plan), and propagate PlanChanged
+	// notifications to all notification subscribers.
+	err = o.planMgr.Load()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Since the plan is now associated with its own manager, I've had to remove the lazy loading that happened in the previous design. We now have (and in the future will have) many managers that each may have APIs which could require them to have a plan. I've decided to have a simple flow of updates from the Plan Manager to the plan change subscribers, so we need to have the Plan Manager get the plan as soon as possible, and update the consumers as soon as possible so they can function.

Reply:
Read the other comment. Thanks. Resolving this comment.